R/writeMSData.R

```diff
@@ -75,17 +75,23 @@
             hdr <- fData(msData)
             ## o Re-calculate stuff we don't have or which might have been
             ##   changed:
+            ##   Add also lowMZ and highMZ; these are missing in CDF
+            ##   files (issue #250).
             new_vals <- do.call(rbind, lapply(pks, function(sp) {
                 max_pos <- base::which.max(sp[, 2])[1]
                 cbind(peaksCount = nrow(sp),
                       totIonCurrent = sum(sp[, 2], na.rm = TRUE),
                       basePeakMZ = sp[max_pos, 1][1],
-                      basePeakIntensity = sp[max_pos, 2])
+                      basePeakIntensity = sp[max_pos, 2],
+                      lowMZ = min(sp[, 1]),
+                      highMZ = max(sp[, 1]))
```

Comment:
Isn't `lowMZ = sp[1, 1]` and `highMZ = sp[nrow(sp), 1]`? Or can't we assume correct order at this stage? (I am not sure anymore but if I remember correctly you ensure ordering of mz values in the ctor for spectra.)

Reply:
You are right - that simplifies it (assuming correct ordering of m/z, but if I'm not mistaken that is also checked in the validity method of `Spectrum` objects).